internal/http.go

```diff
@@ -48,6 +49,50 @@ func (h *basicAuthHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	h.next.ServeHTTP(rw, req)
 }
 
+type TokenAuthErrorFunc func(rw http.ResponseWriter)
+
+// TokenAuthHandler returns a http handler that requires `Authorization: Token <token>`
+// Introduced to support InfluxDB 2.x style authentication
+// https://v2.docs.influxdata.com/v2.0/reference/api/#authentication
+func TokenAuthHandler(token string, onError TokenAuthErrorFunc) func(h http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return &schemeAuthHandler{
+			scheme:      "Token",
+			credentials: token,
+			onError:     onError,
+			next:        h,
+		}
+	}
+}
+
+// General auth scheme handler - match `Authorization: <scheme> <credentials>`
+type schemeAuthHandler struct {
+	scheme      string
+	credentials string
+	onError     TokenAuthErrorFunc
+	next        http.Handler
+}
+
+func (h *schemeAuthHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	if h.credentials != "" {
+		// Scheme checking
+		authHeader := req.Header.Get("Authorization")
+		authParts := strings.SplitN(authHeader, " ", 2)
+		if len(authParts) != 2 ||
+			subtle.ConstantTimeCompare(
+				[]byte(strings.ToLower(strings.TrimSpace(authParts[0]))),
+				[]byte(strings.ToLower(h.scheme))) != 1 ||
+			subtle.ConstantTimeCompare([]byte(strings.TrimSpace(authParts[1])), []byte(h.credentials)) != 1 {
```

Comment:
Let's remove the normalization steps here, just do a subtle.ConstantTimeCompare against the header value.